File: lib/core/services/cour.service.dart
import 'package:gest_school_fluter/core/models/cour.modal.dart';
import 'package:gest_school_fluter/core/repositories/cour.repository.dart';

class CourService {
  static CourRepository courRepository = CourRepository();
  static Future<List<Cour>> findAll() async {
    return await courRepository.findAll();
  }
}

File: src/app/core/services/cour.service.ts
import { Observable } from 'rxjs';
import { CourListe, CreatCour } from '../models/cour.liste';
import { RestResponse } from '../models/rest.response';

export interface CourService {
  findall(
    page: number,
    telephone: string
  ): Observable<RestResponse<CourListe[]>>;
  findallByProfId(
    page: number,
    idprof: string,
    telephone: string
  ): Observable<RestResponse<CourListe[]>>;
  create(creatCour: CreatCour): Observable<RestResponse<CreatCour>>;
}
